content/articles/statistical-hypothesis-testing-python-implementation/index.md

```diff
@@ -0,0 +1,573 @@
+ 
+**The smaller our p-value is, the more delicate it is to accept our null hypothesis.**
+ 
+To calculate our `p-value`, there should be a certain level of significance to the initial assumption.
+
+A smaller p-value means that there are numerous reasons why we should accept the alternative hypothesis.
+ 
+A p-value is usually within the range of 0 and 1. The threshold value is set at 0.05 before the experiments in order to avoid bias.
+
+The alpha level allows us to ascertain whether to accept or reject the null hypothesis. It refers to the probability of making a wrong decision when the ground truth(H0) is true. 
+
+The alpha level is determined by the following formula:
+
+```py
+alpha = 1 - confidence level
+```
+
+If the confidence level from what we've seen in our data is `95%`, then the alpha will be `0.05`:
+
+```py
+alpha = 1 - 0.95
+```
+
+It is important to set the alpha level correctly to avoid biased decisions, especially when we reject the null hypothesis. The standard or preferred level is 5%.
+ 
+### Statistical techniques for hypothesis testing
+The three methods to be discussed are:
+1. Chi-square test
+2. T-test
+3. ANOVA test
+ 
+#### Chi-square test
+Chi-square test is used to analyze two categorical variables in a data set. It focuses on the relationship between these two categorical variables. For example, comparing the after effects of malaria drug A to those of malaria drug B.
+
+Chi-square is a tool for checking how divergent the observed frequency is from the expected results.
+
+In a Chi-square test, we accept the null hypothesis when variable A and variable B are not dependent or have no relationship between them.
+
+We also use the alternate hypothesis when variable A depends on variable B. 
+
+The formula for calculating Chi-square test is shown below:
+ 
+![chi_square_formula](/engineering-education/statistical-hypothesis-testing-python-implementation/chi-square.png)
+
+Consider the following example:
+ 
+A mathematics teacher conducted a test for a class of 20 students. He expected 5 to score good marks, 7 to score average marks while 8 to fail. Eventually, when the result came out, 5 failed, 6 were average while 9 scored good marks. 
+
+We can use a Chi-square test to analyze the performance of this class.
+ 
+|  | Good  |  Average  | Fail | Total |
+|----|--------|:----------|--------------|--------|
+| Observed| 9 | 6 | 5 | 20 |
+| Expected| 5 | 7 | 8 | 20 |
+
+From the table above, students fall into three categories ranging from what is expected to what is finally recorded. The main categories are `good`, `average`, and `fail`. 
+
+Let's denote the expected and observed values with letters to make it more understandable.
+
+Let the: 
+- Observed good be `A` and the expected good be `a`.
+- Observed average be `B` and the expected average be `b`.
+- Observed fail be `C` and the expected to fail be `c`.
```

Comment:
The "O"s in "Observed" shouldn't be capitalized. Please change them to small "o"s.